api/v1alpha1/imagepolicy_types.go

```diff
@@ -55,6 +63,33 @@ type SemVerPolicy struct {
 	Range string `json:"range"`
 }
 
+// AlphabeticalPolicy specifices a alphabetical ordering policy.
+type AlphabeticalPolicy struct {
+	// Order specifies the sorting order of the tags. Given the letters of the
+	// alphabet as tags, ascending order would select Z, and descending order
+	// would select A.
+	// +kubebuilder:default:="asc"
+	// +kubebuilder:validation:Enum=asc;desc
+	// +optional
+	Order string `json:"order,omitempty"`
+}
+
+// TagPrefixMatcher enables filtering tags based on a set of defined rules
+type TagPrefixMatcher struct {
```

Comment:
I like how this is expressed. At risk of exploding the scope, is it ever useful to match by suffix? Or using a glob? (I would count regexp out, per xkcd).

Reply:
Considering there are images that have suffixes such as `-alpine`, `-slim`, `-rc` etc. it might be a valid use case. I'm not sure what's the best approach around dealing with both of these use cases, I kind of like more the idea of globbing patterns vs prefix/suffix matching, would probably make sense.

Reply:
Using something like [this](https://github.com/gobwas/glob#example) maybe?

The only downside moving to glob patterns would probably be that we wouldn't be able to support the current pre-sort "trimming" functionality.

I'm trying to figure out, as a user, what would bring in the most value at the moment.

Reply:
Hmm yeah glob libraries don't tend to implement captures.

Meta-suggestion: shall we move this discussion to an issue or new PR, and reduce this PR to just the alphabetical ordering and related refactoring? That would still be pretty helpful, e.g., for calendar-versioned images.

Reply:
I agree, will move out the prefix filtering to a separate branch and open a PR once this one gets merged.

Reply:
@relu i see your branch for this, but do not see an open PR.  Dont want to step on any toes, but can we open this PR? Our currently workflow prevents us from moving to the Helm Controller until we have this functionality, but the helm operator is breaking due to the deprecation of the old stable repo, so we are stuck currently: https://github.com/fluxcd/image-reflector-controller/tree/tag-prefix-matching